python/lsst/analysis/tools/atools/refCatMatchPlots.py

```diff
@@ -529,3 +533,73 @@ def setDefaults(self):
 
         self.produce.plotName = "astromDiffSky_Dec"
         self.produce.zAxisLabel = "Dec$_{{target}}$ - Dec$_{{ref}}$ (marcsec)"
+
+
+class TargetRefCatDeltaVisitMetrics(TargetRefCatDelta):
+    """Calculate the AA1, AB1, and ABF1 metrics from the difference between the
+    target and reference catalog coordinates.
+    """
+
+    AB2 = Field[float](
+        doc="Distance (in mas) from the mean at which to calculate percentage of outliers", default=20
+    )
+    coordinate = ChoiceField[str](
+        doc="The coordinate direction",
+        optional=False,
+        allowed={"RA": "Right ascension", "Dec": "Declination"},
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        self.applyContext(VisitContext)
+        self.applyContext(RefMatchContext)
+
+        self.process.buildActions.astromDiff = ConvertUnits(
+            buildAction=SubtractVector, inUnit="degree", outUnit="milliarcsecond"
+        )
+
+        self.process.buildActions.mags = ConvertFluxToMag()
+        self.process.buildActions.mags.vectorKey = "psfFlux_target"
+
+        self.process.filterActions.brightStars = DownselectVector(vectorKey="astromDiff")
+        self.process.filterActions.brightStars.selector = RangeSelector(
+            vectorKey="mags", minimum=17, maximum=21.5
+        )
+
+        self.process.calculateActions.AA1 = MedianAction(vectorKey="brightStars")
+        self.process.calculateActions.AB1 = SigmaMadAction(vectorKey="brightStars")
+
+        self.produce.metric.units = {"AA1": "mas", "AB1": "mas", "ABF1": "percent"}
+
+    def finalize(self):
+        super().finalize()
+
+        if self.coordinate == "RA":
+            self.process.buildActions.astromDiff.buildAction.actionA = RAcosDec(
+                raKey="coord_ra_target", decKey="dec_ref"
+            )
+            self.process.buildActions.astromDiff.buildAction.actionB = RAcosDec(
+                raKey="ra_ref", decKey="dec_ref"
+            )
+
+        else:
+            self.process.buildActions.astromDiff.buildAction.actionA = LoadVector(
+                vectorKey="coord_dec_target"
+            )
+            self.process.buildActions.astromDiff.buildAction.actionB = LoadVector(vectorKey="dec_ref")
+
+        self.process.calculateActions.ABF1 = FracThreshold(
+            vectorKey="brightStars",
+            threshold=self.AB2,
+            percent=True,
+            op="gt",
+            relative_to_median=True,
+            absolute_value=True,
+        )
```

Comment:
I also think this should be calculated using separations in the visit's band relative to separations in the r-band. (i.e., it requires a pair of visits -- one in a non-r band, and one in r-band)

Reply:
Moved to DM-39256.